src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -717,21 +717,23 @@ private void ParseHelper(string[] args)
                     _noExit = false;
                     break;
                 }
-                else if (MatchSwitch(switchKey, "help", "h") || MatchSwitch(switchKey, "?", "?"))
+
+                if (MatchSwitch(switchKey, "help", "h") || MatchSwitch(switchKey, "?", "?"))
                 {
                     _showHelp = true;
                     _showExtendedHelp = true;
                     _abortStartup = true;
                 }
+                else if (MatchSwitch(switchKey, "login", "l"))
```

Comment:
Ideally we should move this to the bottom, since the check will never succeed and all switches below it will check this first.

But I left it here because of alpha ordering and also because I feel like it might be worth doing a wider investigation of the switch detection logic for startup performance.